### PR TITLE
Fix: don't try to strip "line:" prefix from parser errors with no such prefix

### LINKED ISSUE #2698: eslint mangles error messages thrown by parser which don't start with "line \d+:"
Custom parsers may throw error messages which don't start with e.g. "line 123:". The [current code][0] just strips off everything before the first colon in the error message, which leads to a mangled message if it didn't actually start with "line XX:".

[0]: https://github.com/eslint/eslint/blob/c8a56e2b7c3f27331f93c607947e8c18caa16c18/lib/eslint.js#L468

I'm working on this, but the tests depend on #2694 being merged so I'll wait until that's closed before actually submitting my PR.

### PR DESCRIPTION
**Pull request by gfxmonk**
fixes #2698

### CODE REVIEW

File: tests/lib/eslint.js [hunk 1]
```diff
@@ -2437,6 +2437,9 @@ describe("eslint", function() {
 
         describe("Custom parser", function() {
 
+            var path = require("path");
+            var parserFixtures = path.join(path.dirname(module.id), "../fixtures/parsers") + "/";
```
Comment (gfxmonk) [inline tests/lib/eslint.js]: I don't know whether this will work correctly on Windows. Anyone have a better idea?

Comment (gyandeeps) [inline tests/lib/eslint.js]: why not just do:
```js
var parserFixtures = path.resolve(__dirname, "../fixtures/parsers");
```

File: tests/lib/eslint.js [hunk 2]
```diff
@@ -2451,6 +2454,22 @@ describe("eslint", function() {
                 assert.equal(messages[0].message, "Cannot find module 'esprima-fbxyz'");
             });
 
+            it("should strip leading line: prefix from parser error", function() {
+                var parser = parserFixtures + "line-error.js";
```
Comment (gyandeeps) [inline tests/lib/eslint.js]: just to be safe why not use `path.join` here since you already have path in context.

File: tests/lib/eslint.js [hunk 3]
```diff
@@ -2437,6 +2437,9 @@ describe("eslint", function() {
 
         describe("Custom parser", function() {
 
+            var path = require("path");
```
Comment (gyandeeps) [inline tests/lib/eslint.js]: You can actually declare this globally.

File: lib/eslint.js [hunk 4]
```diff
@@ -461,12 +461,18 @@ module.exports = (function() {
             });
         } catch (ex) {
 
+            // If the message includes a leading line number, strip it:
+            var message = ex.message;
+            var lineMarker = message.match(/^line \d+: */i);
```
Comment (nzakas) [inline lib/eslint.js]: Is that a repeated space in the regex? I'd suggest leaving that off and then trim the result to make it clearer.

**Comment by gfxmonk** [comment]
Thanks for the tips @gyandeeps, fixed.

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**